pkg/controllers/cloudmap_controller.go

```diff
@@ -80,15 +80,20 @@ func (r *CloudMapReconciler) Reconcile(ctx context.Context) error {
 }
 
 func (r *CloudMapReconciler) reconcileNamespace(ctx context.Context, namespaceName string) error {
-	r.Logger.Info("syncing namespace", "namespace", namespaceName)
-
 	desiredServices, err := r.Cloudmap.ListServices(ctx, namespaceName)
 	if err != nil {
 		return err
 	}
+	// Reconcile applicable for namespaces which have corresponding Cloudmap's namespace and services
```

Comment:
What happens in the case where all the service imports are deleted in Cloud Map? Wouldn't we still need to  clean up the existing imports?

Reply:
We have a task to take care of imports from the cloudmap https://github.com/aws/aws-cloud-map-mcs-controller-for-k8s/issues/21

Reply:
#21 is not related, it's for modifying endpoints on existing imports.
Needing to wipe out all imports is a good callout, I'll add a task for 0.2 since it's a small change.

Reply:
So, we need to add a separate task for reconciling imported `Service`?

Reply:
So, we need to add a separate task for reconciling imported `Service`?

Reply:
Service Imports are pruned if they don't exist in Cloud Map, but with this change [we don't iterate over the existingImportsMap](https://github.com/aws/aws-cloud-map-mcs-controller-for-k8s/blob/main/pkg/controllers/cloudmap_controller.go#L113) if there are no services.
I missed that this was introduced by this PR, so it should not be a separate task, we need to prune imports if there are no services in Cloud Map.

Reply:
I think removing the if check on `len(desiredServices) == 0` should be sufficient to address this